libraries/lib-components/EffectInterface.h

```diff
@@ -481,6 +474,13 @@ class COMPONENTS_API EffectInstance
    //! Function that has not yet found a use
    //! Correct definitions of it will likely depend on settings and state
    virtual size_t GetTailSize() const;
+
+   //! Main thread receives updates to settings from a processing thread
+   /*!
+    Default implementation simply assigns by copy, not move
+    This might be overridden to copy contents only selectively
+    */
```

Comment:
Why do you require R-vaue here in case you expect copy assignment?

Reply:
Allowing more freedom to overriding methods if it is needed

Reply:
You force very harsh contract on the users of this code: they have to provide R-value to this function.